crates/modelardb_server/src/parser.rs

```diff
@@ -875,58 +876,61 @@ mod tests {
-            // skipped in this test.
-            if keyword == &"END-EXEC" {
-                continue;
-            }
-            let keyword_lowercase = keyword.to_lowercase();
-            let error = tokenize_and_parse_sql(
-                format!(
-                    "CREATE MODEL TABLE {}(timestamp TIMESTAMP, field FIELD, tag TAG)",
-                    keyword_lowercase
-                )
-                .as_str(),
-            );
+    fn test_tokenize_and_parse_create_table_with_lowercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_lowercase,
+            "CREATE TABLE {}(timestamp TIMESTAMP, values REAL, metadata REAL)",
+        )
+    }
 
-            assert!(error.is_err());
+    #[test]
+    fn test_tokenize_and_parse_create_table_with_uppercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_uppercase,
+            "CREATE TABLE {}(timestamp TIMESTAMP, values REAL, metadata REAL)",
+        )
+    }
 
-            assert_eq!(
-                error.unwrap_err(),
-                ParserError::ParserError(format!(
-                    "Reserved keyword '{}' cannot be used as a table name.",
-                    keyword_lowercase
-                ))
-            );
-        }
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_with_lowercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_lowercase,
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, field FIELD, tag TAG)",
+        )
     }
 
     #[test]
     fn test_tokenize_and_parse_create_model_table_with_uppercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_uppercase,
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, field FIELD, tag TAG)",
+        )
+    }
+
+    fn parse_and_assert_if_keyword_is_restricted_in_table_name(
+        keyword_to_table_name: fn(&str) -> String,
+        sql: &str,
+    ) {
         for keyword in ALL_KEYWORDS {
             // END-EXEC cannot be parsed by the SQL parser because of the hyphen, and is therefore
             // skipped in this test.
             if keyword == &"END-EXEC" {
                 continue;
             }
-            let keyword_uppercase = keyword.to_uppercase();
-            let error = tokenize_and_parse_sql(
-                format!(
-                    "CREATE MODEL TABLE {}(timestamp TIMESTAMP, field FIELD, tag TAG)",
-                    keyword_uppercase
-                )
-                .as_str(),
+            let statement = tokenize_and_parse_sql(
+                sql.replace("{}", keyword_to_table_name(keyword).as_str())
+                    .as_str(),
             );
 
+            let error = semantic_checks_for_create_table(&statement.unwrap());
```

Comment:
Minor but maybe rename `error` to something like `result` since it is not guaranteed to be an error. 
`assert!(result.is_err());` is also used throughout the rest of the code.